pkg/cmd/diagnostics/diagnostics_download.go

```diff
@@ -0,0 +1,51 @@
+package diagnostics
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/dcos/dcos-cli/api"
+	diagnostics "github.com/dcos/dcos-core-cli/pkg/diagnostics/v2"
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
+	"github.com/spf13/cobra"
+)
+
+func newDiagnosticsDownloadCommand(ctx api.Context) *cobra.Command {
+	var outputPath string
+
+	cmd := &cobra.Command{
+		// the <bundle-id> is used to indicate what the expected positional arg is in the help output
+		Use:   "download <bundle-id>",
+		Short: "Download diagnostics bundle",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := pluginutil.HTTPClient("")
+			client := diagnostics.NewClient(c)
+
+			id := args[0]
+
+			if outputPath == "" {
+				outputPath = fmt.Sprintf("%s.zip", id)
```

Comment:
I think we should sanitize this user input in some way to prevent malicious behaviour.

Reply:
We could limit `id` to contain only alphanumerics and dashes but I'm not sure what we could do to `outputPath` to prevent issues. Maybe not allow it to have `..` but if we're worried about malicious use, an attacker could just as easily `cd` to their target directory first (or `/`).

Maybe I'm not seeing the potential malicious behavior you are.

Reply:
It looks fine as is IMO, users should be able to set any path as long as they have permission to create a file there.

Reply:
However I wouldn't add the `.zip` suffix automatically. If I pass an explicit flag I wouldn't expect this behavior. That's similar to what we do in other commands, eg. `dcos license audit get -o /path/to/file`

Reply:
@br-lewis, you're probably right. In the end the file will be created on the user's filesystem, so he could potentially only harm himself.